doc/reference/best_practices.md

```diff
@@ -0,0 +1,25 @@
+# Best Practices
+
+This document discusses best practices for common use cases.
+
+## Shuffling files
+
+Certain pipelines simply shuffle files around.  If you find yourself writing a pipeline that does a lot of copying, it probably falls into this category.
```

Comment:
If you find yourself writing a pipeline that does a lot of copying, such as [Time Windowing] (http://docs.pachyderm.io/en/latest/cookbook/time_windows.html), it probably falls into this category.